### PR TITLE
PEO-591 [CircleCi] Introduction steps refactoring

### PR DESCRIPTION
**Pull request by D3N**
[PEO-591](https://garage.atlassian.net/browse/PEO-591)

### CODE REVIEW

File: intro_to_circleci.md [hunk 1]
```diff
@@ -485,30 +485,166 @@ Caches are isolated based on GitHub Repo for PRs. CircleCI uses the GitHub repos
...
+
+  - ```at``` - directory to attach the workspace to
+
+- **```add_ssh_keys```**
+
+  Special step that adds SSH keys from a project’s settings to a container. Also configures SSH to use these keys. Subkey:
+
+  - ```fingerprints``` - list of fingerprints corresponding to the keys to be added (default: all keys added)
+
+There is also an ability to set up custom reusable steps as commands:
```
Comment (kolyanjy) [inline intro_to_circleci.md]: could be replaced with a more life-like example

File: intro_to_circleci.md [hunk 2]
```diff
@@ -485,30 +485,167 @@ Caches are isolated based on GitHub Repo for PRs. CircleCI uses the GitHub repos
...
+
+    - ```name``` - title of the step to be shown in UI
+    - ```shell``` - shell to use for execution command
+    - ```environment``` - additional environmental variables, locally scoped to command
+    - ```background``` - configures commands to run in the background (default: false)
+
+      The following example shows the config for running the X virtual framebuffer in the background  which is commonly required to run Selenium tests:
+      ```yml
+      - run:
+          name: Running X virtual framebuffer
```
Comment (dzemlianoi-double) [inline intro_to_circleci.md]: wtf. it's toooo hard

File: intro_to_circleci.md [hunk 3]
```diff
@@ -485,30 +485,167 @@ Caches are isolated based on GitHub Repo for PRs. CircleCI uses the GitHub repos
...
+
+  A conditional step consists of a step with the key ```when``` or ```unless```. The purpose of the ```when``` step is customizing commands and job configuration to run on custom conditions (determined at config-compile time) that are checked before a workflow runs. Under the ```when``` key are the subkeys:
+    - ```condition``` - a parameter value
+    - ```steps``` - a list of steps to execute when the condition is true
+
+    ```yml
+    jobs:
+      steps:
+        - when:
+            condition: <<parameters.custom_checkout>>
```
Comment (dzemlianoi-double) [inline intro_to_circleci.md]: where this parameters could be sent

File: intro_to_circleci.md [hunk 4]
```diff
@@ -485,30 +485,167 @@ Caches are isolated based on GitHub Repo for PRs. CircleCI uses the GitHub repos
...
+
+- **```restore_cache```**
+
+- **```deploy```**
+
+  Special step for deploying artifacts. ```deploy``` step behaves and uses the same configuration map and semantics as ```run``` step. Jobs may have more than one ```deploy``` step.
+
+- **```store_artifacts```**
+
+  Step to store artifacts (for example logs, binaries, etc) to be available in the web app or through the API. Subkeys:
```
Comment (dzemlianoi-double) [inline intro_to_circleci.md]: logs, binaries, failed feature tests screenshots

File: intro_to_circleci.md [hunk 5]
```diff
@@ -485,30 +485,167 @@ Caches are isolated based on GitHub Repo for PRs. CircleCI uses the GitHub repos
...
+        - vendor/bundle
+```
+
+```yml
+steps:
+  - <<: *restore_bundle_cache
+  - <<: *bundle_install
+  - <<: *save_bundle_cache
+```
+In example above were used YAML anchors & aliases.
```
Comment (dzemlianoi-double) [inline intro_to_circleci.md]: Best practises

**Pull request merged**